src/services/webdav/dir_stream.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::Result;
+use crate::{raw::output, ObjectMetadata, ObjectMode};
+use async_trait::async_trait;
+
+use super::list_response::Multistatus;
+
+pub struct DirStream {
+    size: usize,
+    multistates: Multistatus,
+}
+
+impl DirStream {
+    pub fn new(multistates: Multistatus, limit: Option<usize>) -> Self {
+        Self {
+            size: limit.unwrap_or(1000),
+            multistates,
+        }
+    }
+}
+
+#[async_trait]
+impl output::Page for DirStream {
+    async fn next_page(&mut self) -> Result<Option<Vec<output::Entry>>> {
```

Comment:
@Xuanwo sorry, I found some logic errors here but am not sure how to fix it.

```
Handling dir like start a new list via meta.path()
de: "/"
Handling dir like start a new list via meta.path()
de: "/"
Handling dir like start a new list via meta.path()
de: "/"
Handling dir like start a new list via meta.path()
de: "/"
Handling dir like start a new list via meta.path()
de: "/"
Handling dir like start a new list via meta.path()
```

Looks like it is an infinite loop?